diff --git a/orocos_kdl/src/CMakeLists.txt b/orocos_kdl/src/CMakeLists.txt
index 079ca8a..cea55fb 100644
--- a/orocos_kdl/src/CMakeLists.txt
+++ b/orocos_kdl/src/CMakeLists.txt
@@ -133,6 +133,7 @@ INSTALL(TARGETS orocos-kdl
   EXPORT OrocosKDLTargets
   ARCHIVE DESTINATION lib${LIB_SUFFIX}
   LIBRARY DESTINATION lib${LIB_SUFFIX}
+  RUNTIME DESTINATION bin
   PUBLIC_HEADER DESTINATION include/kdl
 )
 
