diff --git a/python_orocos_kdl/cmake/SIPMacros.cmake b/python_orocos_kdl/cmake/SIPMacros.cmake
index 15b0c5a..c07b5ad 100644
--- a/python_orocos_kdl/cmake/SIPMacros.cmake
+++ b/python_orocos_kdl/cmake/SIPMacros.cmake
@@ -115,7 +115,9 @@ MACRO(ADD_SIP_PYTHON_MODULE MODULE_NAME MODULE_SIP)
     ELSE (CYGWIN OR APPLE)
         ADD_LIBRARY(${_logical_name} SHARED ${_sip_output_files} )
     ENDIF (CYGWIN OR APPLE)
-    TARGET_LINK_LIBRARIES(${_logical_name} ${PYTHON_LIBRARY})
+    IF (NOT APPLE)
+        TARGET_LINK_LIBRARIES(${_logical_name} ${PYTHON_LIBRARY})
+    ENDIF (NOT APPLE)
     TARGET_LINK_LIBRARIES(${_logical_name} ${EXTRA_LINK_LIBRARIES})
     SET_TARGET_PROPERTIES(${_logical_name} PROPERTIES PREFIX "" OUTPUT_NAME ${_child_module_name})
 

